### PR TITLE
add new repo structure

### LINKED ISSUE #3: [COWPY-2] Setup base repo structure
* add base python package;
* add base CI;
* add formatter;
* move some of the functions from commons/subgraphs package from [https://github.com/cowprotocol/cow-sdk](https://github.com/cowprotocol/cow-sdk)

<sub>From [SyncLinear.com](https://synclinear.com) | [COWPY-2](https://linear.app/bleu-llc/issue/COWPY-2/setup-base-repo-structure)</sub>

### PR DESCRIPTION
**Pull request by ribeirojose**
closes https://github.com/bleu-fi/cow-py/issues/3

### CODE REVIEW

File: cow_py/core/__init__.py [hunk 1]
```diff
@@ -0,0 +1,4 @@
+from cow_py.core import constants, cow_error, abis, chains, config
```
Comment (yvesfracari) [inline cow_py/core/__init__.py]: would use the same name of the ts package (common)

File: cow_py/core/chains/__init__.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    Chain.MAINNET: "ethereum",
+    Chain.GNOSIS: "gnosis",
+    Chain.SEPOLIA: "sepolia",
+}
+
+CHAIN_SCANNER_MAP = {
+    Chain.MAINNET: "https://etherscan.io",
+    Chain.GNOSIS: "https://gnosisscan.io",
+    Chain.SEPOLIA: "https://sepolia.etherscan.io/",
+}
```
Comment (yvesfracari) [inline cow_py/core/chains/__init__.py]: i don't think that we will need this

File: README.md [hunk 3]
```diff
@@ -1,4 +1,5 @@
 # cow-py
+
```
Comment (yvesfracari) [inline README.md:2]: This was made by the formatter?

**Review by yvesfracari — APPROVED** [review]
lgtm!

File: cow_py/core/constants.py
Comment (mfw78) [inline cow_py/core/constants.py:1]: FYI, on this, I think we would at some stage want to change the upstream cow-sdk to have `contracts` as a module within, so you could have:
* contracts module
* within the contracts module, the abi.

So all consumers of the cow-py don't have to worry about ABI encoding etc (ie. shouldn't have to touch it at all, and the SDK can do it for them).

I won't hold up the merge on this, and happy for you to roll forward :roller_skate:

**Review by mfw78 — APPROVED** [review]

**Review by mendesfabio — APPROVED** [review]

File: cow_py/core/constants.py
Comment (ribeirojose) [reply cow_py/core/constants.py:1]: agreed! Just sent a commit with those changes 😄

File: cow_py/core/chains/__init__.py (hunk 2 above)
Comment (ribeirojose) [reply cow_py/core/chains/__init__.py]: I'll leave it here for now though

**Pull request merged**